runners/flink/flink_runner.gradle

```diff
@@ -169,6 +170,17 @@ dependencies {
   validatesRunner project(path: ":runners:core-java", configuration: "testRuntime")
   validatesRunner project(project.path)
   miniCluster "org.apache.flink:flink-runtime-web_2.11:$flink_version"
+  compile project(path: ":model:fn-execution", configuration: "shadow")
+  compile project(path: ":model:pipeline", configuration: "shadow")
+  compile project(path: ":model:job-management:", configuration: "shadow")
+  compile project(":sdks:java:fn-execution")
+  compile project(path: ":vendor:sdks-java-extensions-protobuf", configuration: "shadow")
+  permitUnusedDeclared library.java.jackson_annotations
+  compile library.java.jackson_databind
+  compile "org.apache.flink:flink-annotations:$flink_version"
+  compile "org.apache.flink:flink-optimizer_2.11:$flink_version"
+  permitUnusedDeclared "org.apache.flink:flink-clients_2.11:$flink_version"
```

Comment:
I think `org.apache.flink:flink-clients_2.11:$flink_version` should be changed to a `testCompile` dependency instead.

Reply:
testCompile does not work here, flink-clients has to be UnusedDeclared.

Reply:
What is the error if we remove the `compile` dependency on `flink-clients`?

Ideally every `permitUnusedDeclared` should have a comment explaining why the dependency is actually needed.

Reply:
Same here - since `compile "org.apache.flink:flink-clients_2.11:$flink_version"` could safely be removed, we no longer need `permitUnusedDeclared "org.apache.flink:flink-clients_2.11:$flink_version"`.